mozilla_schema_generator/common_ping.py

```diff
@@ -12,23 +12,26 @@
 from typing import List
 
 
-class MainPing(GenericPing):
+class CommonPing(GenericPing):
 
     # Only includes probes that have been available at some point past
     # this version.
     # ONLY DECREMENT, or the schema will change in an incompatible way!
     MIN_FX_VERSION = 30
 
-    schema_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/master/schemas/telemetry/main/main.4.schema.json" # noqa E501
     env_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/master/templates/include/telemetry/environment.1.schema.json" # noqa E501
     probes_url = GenericPing.probe_info_base_url + "/firefox/all/main/all_probes"
 
-    def __init__(self):
-        super().__init__(self.schema_url, self.env_url, self.probes_url)
+    def __init__(self, schema_url):
```

Comment:
The main difference between `MainPing` and `CommonPing` is that `CommonPing` can process different schemas

Reply:
I'm using `MainPing` in the upcoming probe search service to collect the non-glean pings for desktop Firefox. After this change it looks like I would need to provide the full schema URL? The `main.4.schema.json` one?

Reply:
It might make sense to subclass the MainPing as a type of CommonPing, and to move modifications to `payload.simple_measurements` there. The URL could be fixed there, leaving the API backwards compatible.

Reply:
I added `MainPing` back as subclass of `CommonPing`. So it should be backwards compatible now. Also `payload.simple_measurements` will now only get set for main ping

Reply:
Great, thank you